ethcore/src/state/mod.rs

```diff
@@ -519,8 +519,14 @@ impl State {
 
 		// TODO uncomment once to_pod() works correctly.
 //		trace!("Applied transaction. Diff:\n{}\n", state_diff::diff_pod(&old, &self.to_pod()));
-		self.commit()?;
-		let receipt = Receipt::new(self.root().clone(), e.cumulative_gas_used, e.logs);
+		let state_root = match env_info.number < engine.params().eip98_transition {
```

Comment:
Using `if-else` here would a bit more idiomatic and suprisingly shorter then `match`.